### DISCUSSION TITLE
How to test pushing to GitHub Package Registry via a GitHub Action?

### DISCUSSION THREAD

**Discussion by julie-sullivan** [community/community#25077] [Actions]
<p>I’ve pushed my 1.0 JAR to the GitHub package registry, works great!</p>
<p>I’ve now written a GitHub Action to publish automatically to the registry when I create a tag.</p>
<p>How can I test this action? Since packages are immutable, I am afraid to make a mistake and have a bad JAR I can’t ever delete.</p>
<p>This is the untested part of my action that I am hoping works:</p>
<pre><code>run: |
mkdir ~/.m2
echo "&lt;settings&gt;&lt;servers&gt;&lt;server&gt;&lt;id&gt;github&lt;/id&gt;&lt;username&gt;USERNAME&lt;/username&gt;&lt;password&gt;${GITHUB_TOKEN}&lt;/password&gt;&lt;/server&gt;&lt;/servers&gt;&lt;/settings&gt;" &gt; ~/.m2/settings.xml
mvn deploy
</code></pre>
<p>And the corresponding repository information is in the pom.xml file.</p>
<p>Suggestions?</p>

**Comment by chrispat** [comment, marked as answer]
<p>The setup-java action has detals on publishing maven packages to the package registry <a href="https://github.com/actions/setup-java#publishing-using-apache-maven" rel="nofollow noopener">https://github.com/actions/setup-java#publishing-using-apache-maven</a>.  As far as testing goes I would recommend creating a different repo if you want to delete everything after your publishing tests are complete.</p>
